### PR TITLE
feat(lambda-event-sources): add filters to Amazon MSK, Self-Managed Kafka event sources

### PR DESCRIPTION
**Pull request by nikovirtala**
This pull request adds support for content filtering options for Amazon MSK and Self-Managed Kafka as announced on: https://aws.amazon.com/about-aws/whats-new/2022/10/aws-lambda-event-filtering-amazon-msk-kafka-mq-apache-activemq-amazon-mq-rabbit-mq/

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nikovirtala** [comment]
Ok, @aws-cdk-automation. How do you want me to change the `README.md`? The current version doesn't say a word about the filtering, even for the already supported event sources 🤷🏼

**Comment by Naumel** [comment]
> Ok, @aws-cdk-automation. How do you want me to change the `README.md`? The current version doesn't say a word about the filtering, even for the already supported event sources 🤷🏼

Don't throw rhetorical questions into the void 😄
Edits are not the only way to bring changes to a README file, do you really think nothing should be added?

**Comment by nikovirtala** [comment]
> Edits are not the only way to bring changes to a README file, do you really think nothing should be added?

Quite the opposite. I'm after answer whether I should add a generic paragraph about filtering or something Kafka specific, like the pull request?

**Comment by Naumel** [comment]
> > Edits are not the only way to bring changes to a README file, do you really think nothing should be added?
> 
> Quite the opposite. I'm after answer whether I should add a generic paragraph about filtering or something Kafka specific, like the pull request?

My knee-jerk reaction is that if it's easy to search and find good result, generic becomes superfluous.
In this case looks like more specific would fit better, does that sound right to you?

**Comment by TheRealAmazonKendra** [comment]
> > > Edits are not the only way to bring changes to a README file, do you really think nothing should be added?
> > 
> > 
> > Quite the opposite. I'm after answer whether I should add a generic paragraph about filtering or something Kafka specific, like the pull request?
> 
> My knee-jerk reaction is that if it's easy to search and find good result, generic becomes superfluous. In this case looks like more specific would fit better, does that sound right to you?

Looks like we had a difference of opinion here and I commented before seeing @Naumel's comment. I would say that specific is the more important section, but if it requires a general explanation of filtering to make sense, that should also be added.

File: packages/@aws-cdk/aws-lambda-event-sources/test/integ.kafka-selfmanaged.ts [hunk 1]
```diff
@@ -44,6 +44,14 @@ zp2mwJn2NYB7AZ7+imp0azDZb+8YG2aUCiyqb6PnnA==
       'my-self-hosted-kafka-broker-3:9092',
     ];
 
+    const filters = [
```
[the remark this answers is not in the document]
Comment (nikovirtala) [reply packages/@aws-cdk/aws-lambda-event-sources/test/integ.kafka-selfmanaged.ts:47]: I can add a new test, but your code is not guiding me by setting an example – all the existing integration tests are part of this one test.

File: packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts [hunk 2]
```diff
@@ -29,6 +29,12 @@ export interface KafkaEventSourceProps extends BaseStreamEventSourceProps {
    * @default - none
    */
   readonly consumerGroupId?: string;
+  /**
+   * Add filter criteria option
+   *
+   * @default - None
+   */
+  readonly filters?: Array<{[key: string]: any}>;
```
[the remark this answers is not in the document]
Comment (nikovirtala) [reply packages/@aws-cdk/aws-lambda-event-sources/lib/kafka.ts:37]: Maybe I don't know 🤷🏼 

If you can tell based on "Properly filtering Amazon Managed Streaming for Apache Kafka, self-managed Apache Kafka, and Amazon MQ messages" chapter on https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html it is possible, then I can try to implement it.

**Comment by nikovirtala** [comment]
> > > > Edits are not the only way to bring changes to a README file, do you really think nothing should be added?
> > > 
> > > 
> > > Quite the opposite. I'm after answer whether I should add a generic paragraph about filtering or something Kafka specific, like the pull request?
> > 
> > 
> > My knee-jerk reaction is that if it's easy to search and find good result, generic becomes superfluous. In this case looks like more specific would fit better, does that sound right to you?
> 
> Looks like we had a difference of opinion here and I commented before seeing @Naumel's comment. I would say that specific is the more important section, but if it requires a general explanation of filtering to make sense, that should also be added.

So, both. [This](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html) is the source of truth for filtering I guess ... I will try to extract to most crucial parts of here and link to the source for more details? Or is that, "no-no"?

**Comment by TheRealAmazonKendra** [comment]
> So, both. [This](https://docs.aws.amazon.com/lambda/latest/dg/invocation-eventfiltering.html) is the source of truth for filtering I guess ... I will try to extract to most crucial parts of it here and link to the source for more details? Or is that, "no-no"?

That's a good approach

File: packages/@aws-cdk/aws-lambda-event-sources/test/integ.kafka-selfmanaged.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheRealAmazonKendra) [reply packages/@aws-cdk/aws-lambda-event-sources/test/integ.kafka-selfmanaged.ts:47]: See our INTEGRATION_TEST file.

**Pull request closed without merging**